tests/queries/0_stateless/02962_system_sync_replica_lightweight_from_modifier.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+# Tags: zookeeper, no-parallel, no-fasttest
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+# shellcheck source=./replication.lib
+. "$CURDIR"/replication.lib
+
+TOTAL_REPLICAS=10
+REPLICAS_TO_DROP=7
+
+for i in $(seq $TOTAL_REPLICAS); do
+    $CLICKHOUSE_CLIENT --query "DROP TABLE IF EXISTS test_table_$i"
+    $CLICKHOUSE_CLIENT --query "CREATE TABLE test_table_$i (key UInt64, value UInt8) ENGINE = ReplicatedMergeTree('/clickhouse/tables/$CLICKHOUSE_TEST_ZOOKEEPER_PREFIX/test_table', '$i') ORDER BY key"
+done
+
+function insert_thread() {
+    while true; do
+        REPLICA=$(($RANDOM % $TOTAL_REPLICAS + 1))
+        $CLICKHOUSE_CLIENT --query "INSERT INTO test_table_$REPLICA VALUES ($RANDOM, $RANDOM % 255)"
+        sleep 0.$RANDOM
+    done
+}
+
+function sync_and_drop_replicas() {
+    for i in $(seq $REPLICAS_TO_DROP); do
+        $CLICKHOUSE_CLIENT --query "SYSTEM SYNC REPLICA LIGHTWEIGHT test_table_$i FROM '$i'"
+        $CLICKHOUSE_CLIENT --query "DROP TABLE IF EXISTS test_table_$i"
```

Comment:
There's a race condition: a new data part may be inserted after SYNC but before DROP. We should make the replica stop accepting new insert queries before syncing parts from it. Possible ways to do it:
 - add a list of "[un]ready" replicas to the bash script (the easiest way to share state between threads in bash is to use temporary files on disk)
 - `alter table table_to_drop modify setting parts_to_throw_insert=0` or a similar trick